addons/sale/tests/test_sale_order_down_payment.py

```diff
@@ -456,12 +456,37 @@ def test_warning_on_invoice_with_credit_limit(self):
         # that should bring partner_a's credit above its limit.
         self.assertEqual(sale_order.invoice_ids.partner_credit_warning, '')
 
+        invoice = sale_order.invoice_ids
+        invoice.action_post()
+
+        # Create a credit note reversing the invoice
+        move_rev = self.env['account.move.reversal'].with_company(self.env.company).create(
+            {
+                'move_ids': [Command.set((invoice.id,))],
+                'date_mode': 'custom',
+                'journal_id': invoice.journal_id.id
+            }
+        )
+
+        move_rev.reverse_moves()
+
+        credit_note = move_rev.new_move_ids
+        credit_note.ensure_one()
+        credit_note.action_post()
+
+        # Check that the credit note is accounted for correctly for the amount_to_invoice
+        self.assertEqual(sale_order.amount_to_invoice, sale_order.amount_total)
+
+        credit_note.button_draft()
+        credit_note.unlink()
+        invoice.button_draft()
```

Comment:
To avoid doing this, it would be easier to add the credit note at the end of the test (like I suggested initially :innocent:), so it does not interfere with the rest of the flow.